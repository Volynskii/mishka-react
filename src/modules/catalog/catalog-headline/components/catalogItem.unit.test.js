import React, {useRef} from "react";
import {mount} from "enzyme";
import {Provider, default as redux} from "react-redux";
import configureStore from "redux-mock-store";
import renderer, {act} from "react-test-renderer";
import CatalogItem from "./catalogItem";

const basketItems = [
    {id:0,
        src: "./images/catalog/catalog-item-1.png",
        itemPrice: 1200,
        itemName:'Зайчик-попрыгайчик',
        itemParamOneName:'рост',
        itemParamOneValue: 30,
        itemParamOneUnit: 'см',
        itemParamTwoName: 'вес',
        itemParamTwoValue: 200,
        itemParamTwoUnit: 'г',
        itemQuantity: 1
    },
    {id:1,
        src: "./images/catalog/catalog-item-2.png",
        itemPrice: 690,
        itemName:'Корзинка для белья',
        itemParamOneName:'Диаметр',
        itemParamOneValue: 15,
        itemParamOneUnit: 'см',
        itemParamTwoName: 'высота',
        itemParamTwoValue: 10,
        itemParamTwoUnit: 'см',
        itemQuantity: 1
    },
    {id:2,
        src: "./images/catalog/catalog-item-3.png",
        itemPrice: 1500,
        itemName:'Большая корзинка для игрушек',
        itemParamOneName:'Диаметр',
        itemParamOneValue: 30,
        itemParamOneUnit: 'см',
        itemParamTwoName: 'высота',
        itemParamTwoValue: 30,
        itemParamTwoUnit: 'см',
        itemQuantity: 1},

];

const initialState = {
    catalogItems: basketItems,
    basketItems: basketItems,
};
const mockStore = configureStore();
const store = mockStore(initialState);


const initialProps = {
    catalogItem: basketItems,
    basketItem: basketItems,

};

const container = mount(
    <Provider store={store}>
        <CatalogItem {...initialProps}
        />
    </Provider>
);

describe('СatalogItem works', () => {

    it('should render СatalogItem Component', ()=> {
        const dom = container;
        const component = renderer.create(dom);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
