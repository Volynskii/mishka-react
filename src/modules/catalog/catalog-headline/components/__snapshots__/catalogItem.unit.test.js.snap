// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`СatalogItem works should render СatalogItem Component 1`] = `
<li
  className="catalog__list__item"
>
  <div
    className="catalog__list__item__image"
  >
    <img
      alt="картинка товара"
      className="catalog__list__item__image-picture"
    />
  </div>
  <div
    className="catalog__list__item__wrapper"
  >
    <section
      className="catalog__list__item__param"
    >
      <div
        className="catalog__list__item__param__name"
      />
      <div
        className="catalog__list__item__param__description"
      >
         
        , 
         
         
      </div>
    </section>
    <ul
      className="catalog__list__item__order"
    >
      <li
        className="catalog__list__item__order__price"
      >
         р
      </li>
      <li
        className="catalog__list__item__order__basket"
        onClick={[Function]}
      />
    </ul>
  </div>
</li>
`;
